File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPCanPickupItems.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPCanPickupItems implements PlayerStateProperty<Boolean> {
    @Override
    public String getConfigName() {
        return "can_pickup_items";
    }

    @Override
    public Boolean getDefaultSerializedProperty() {
        return true;
    }

    @Override
    public Boolean getSerializedPropertyFromPlayer(Player player) {
        return player.getCanPickupItems();
    }

    @Override
    public void restoreProperty(Player player, Boolean canPickupItems) {
        player.setCanPickupItems(canPickupItems);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPCanPickupItems.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPCanPickupItems extends PlayerStateProperty[Boolean]:
  override def getConfigName: String = "can_pickup_items"

  override def getDefaultSerializedProperty: Boolean = true

  override def getSerializedPropertyFromPlayer(player: Player): Boolean =
    player.getCanPickupItems

  override def restoreProperty(player: Player, canPickupItems: Boolean): Unit =
    player.setCanPickupItems(canPickupItems)